lib/workhorse/db_job.rb

```diff
@@ -12,6 +12,58 @@ class DbJob < ActiveRecord::Base
 
     self.table_name = 'jobs'
 
+    def self.waiting
+      where(state: STATE_WAITING)
+    end
+
+    def self.locked
+      where(state: STATE_LOCKED)
+    end
+
+    def self.started
+      where(state: STATE_STARTED)
+    end
+
+    def self.succeeded
+      where(state: STATE_SUCCEEDED)
+    end
+
+    def self.failed
+      where(state: STATE_FAILED)
+    end
+
+    # Resets job to state "waiting" and clears all meta fields
+    # set by workhorse in course of processing this job.
+    #
+    # This is only allowed if the job is in a "dead-end-state" ("succeeded",
+    # "failed"), as it is safe to do so because workhorse won't touch these
+    # jobs. Set "force" to true in order to reset the job without checking for
+    # its state. If you do so, make sure that the job is not still being
+    # processed by a worker. If possible, shutdown all workers before performing
+    # a forced reset.
```

Comment:
A general remark: I do not particularly like the term "dead-end state", as a (broken) perpetual lock is also a dead end with regards to job execution. Maybe 'performed-state' or 'completion state' is more semantically correct.

Furthermore, a suggestion for editing the comment for clarity:
> This is only allowed if the job is in a "XX state" ("succeeded" or "failed"), as only those jobs are safe to modify; workhorse will not touch these jobs. To reset a job without checking the state it is in, set "force" to true. Prior to doing so, ensure that the job is not still being processed by a worker. If possible, shut down all workers before performing a forced reset.